test/xkeyboard-config-test.py.in

```diff
@@ -54,23 +61,56 @@ class Invocation:
     def rmlvo(self):
         return self.rules, self.model, self.layout, self.variant, self.option
 
-    def __str__(self):
-        s = []
-        rmlvo = [x or "" for x in self.rmlvo]
-        rmlvo = ", ".join([f'"{x}"' for x in rmlvo])
-        s.append(f"- rmlvo: [{rmlvo}]")
-        s.append(f'  cmd: "{escape(self.command)}"')
-        s.append(f"  status: {self.exitstatus}")
+    @staticmethod
+    def rmlvo_labels():
+        return "rules", "model", "layout", "variant", "option"
+
+    @property
+    def rmlvo_yaml(self) -> str:
+        return (
+            "{ "
+            + ", ".join(
+                f'{key}: "{value}"'
+                for key, value in zip(self.rmlvo_labels(), self.rmlvo)
+                if value
+            )
+            + " }"
+        )
```

Comment:
IMO this would be easier to read as 
```suggestion
        kv = ", ".join(
                f'{key}: "{value}"'
                for key, value in zip(self.rmlvo_labels(), self.rmlvo)
                if value
        return f"{{ {kv} }}"
```

with a possible

```py
   f'{key}: "{getattr(self, key)}"' for key in self.rmlvo_labels()
```
so we don't require the order of two 'independent' things to match. though admittedly very niche here